resources/asciidoctor/spec/copy_images_spec.rb

```diff
@@ -31,46 +31,115 @@ def copy_attributes(copied)
 
   spec_dir = File.dirname(__FILE__)
 
-  it "copies a file when directly referenced" do
-    copied = []
-    attributes = copy_attributes copied
-    input = <<~ASCIIDOC
-      == Example
-      image::resources/copy_images/example1.png[]
-    ASCIIDOC
-    convert input, attributes,
-        eq("INFO: <stdin>: line 2: copying #{spec_dir}\/resources\/copy_images\/example1.png")
-    expect(copied).to eq([
-        ["resources/copy_images/example1.png", "#{spec_dir}/resources/copy_images/example1.png"],
-    ])
+  shared_context 'convert' do
+    let(:result) do
+      copied = []
+      attributes = copy_attributes copied
+      result = convert_with_logs input, attributes
+      result.push copied
+      result
+    end
+    let(:logs)   { result[1] }
+    let(:copied) { result[2] }
```

Comment:
I find this to be a little confusing. I think the variables you're interested in can be extracted out into their own `let` blocks and then the `!` can be used to ensure that the result is created before other variables are accessed. I'm not sure if this would work, but I'm thinking something like this that isolates the variables.

```ruby
let!(:logs) do
  attributes = copy_attributes copied
  (convert_with_logs input, attributes)[1]
end

let(:copied) do
  []
end
```

Is the `copied` object mutated by the `copy_attributes` method?